api/node/server/server.go

```diff
@@ -307,3 +314,46 @@ func (s *Server) GetHareBeaconEpoch(ctx context.Context,
 	}
 	return &beaconResp{b: beacon}, nil
 }
+
+type proposalResp struct {
+	buf   []byte
+	nonce types.VRFPostIndex
+}
+
+func (p *proposalResp) VisitGetProposalLayerNodeResponse(w http.ResponseWriter) error {
+	if p.buf == nil {
+		w.WriteHeader(204)
+		return nil
+	}
+	w.Header().Add("content-type", "application/octet-stream")
+	w.Header().Add("x-spacemesh-atx-nonce", fmt.Sprintf("%d", p.nonce))
```

Comment:
so in the current configuration, it isn't clear to me what is persisted on the smesher node and to which degree. I avoided having hard dependencies to the activation service or the "db" because it isn't clear to me what are the guarantees. what happens when the smeshing service restarts? does it have a persistent db or not? due to all of these questions I just delegated this responsibility to the node.

Reply:
The smesher side doesn't preserve it currently. It is part of an ATX and it could be retrieved from the last ATX via `/activation/last_atx/{node_id}` but it _feels_ wrong.

Side note: it's weird to return a part of the response in a header and the other part in the body.

Reply:
> The smesher side doesn't preserve it currently. It is part of an ATX and it could be retrieved from the last ATX via /activation/last_atx/{node_id} but it feels wrong.

Will that be guaranteed to be the same atx which is referenced in the proposal?


> Side note: it's weird to return a part of the response in a header and the other part in the body.

I cannot agree more, but adding more types that use the same sinister polymorphism that's present in the codebase feels even more weird and will further complicate things with more domain types, etc. Since we're using REST, having a header is also fine and legit. But yeah, ideally it is it's own type.